src/file-selector.ts

```diff
@@ -105,23 +105,29 @@ function flatten<T>(items: any[]): T[] {
   );
 }
 
-function fromDataTransferItem(
+async function fromDataTransferItem(
   item: DataTransferItem,
   entry?: FileSystemEntry | null,
-) {
-  if (typeof (item as any).getAsFileSystemHandle === "function") {
-    return (item as any).getAsFileSystemHandle().then(async (h: any) => {
-      const file = await h.getFile();
-      file.handle = h;
-      return toFileWithPath(file);
-    });
+): Promise<FileWithPath> {
+  if (typeof item.getAsFileSystemHandle === "function") {
+    const handle = await item.getAsFileSystemHandle();
+
+    if (!handle || !isFileHandle(handle)) {
+      throw new Error("Transferred item is not a file.");
+    }
+
+    const file = await handle.getFile();
+    (file as any).handle = handle;
+    return toFileWithPath(file);
   }
+
   const file = item.getAsFile();
+
   if (!file) {
-    return Promise.reject(`${item} is not a File`);
+    throw new Error("Transferred item is not a file.");
```

Comment:
Make a func here to return this err (same err in 2 places)? Or make a custom error object with the same msg?

Reply:
Could do that yeah, but seems a bit overkill. I generally use the [Rule of Three](https://en.wikipedia.org/wiki/Rule_of_three_(computer_programming)) for this kind of stuff.